### PR TITLE
Feature/ng toolbar column select

### PR DESCRIPTION
**Pull request by manwithtwowatches**
This still needs to be styled but that can be done in a different branch.

### CODE REVIEW

**Comment by manwithtwowatches** [comment]
https://www.openproject.org/work_packages/7585

File: app/assets/stylesheets/content/_modal.css.sass [hunk 1]
```diff
@@ -14,7 +14,7 @@ $ng-modal-image-width:   $ng-modal-image-height
   +position(fixed, 0px 0px 0px 0px)
   background: rgba(0, 0, 0, 0.85)
   text-align: left
-  z-index: 99999999999
+  z-index: 10000
```
Comment (tilt) [inline app/assets/stylesheets/content/_modal.css.sass]: very modest

**Comment by tilt** [comment]
Looks nice to me, I'll look at it a bit closer soon - some of the stuff might make former code superfluous and I want to move that around or discard it prior to merging this, so that we don't lose track of it.

File: app/views/work_packages/index.html.erb [hunk 2]
```diff
@@ -46,6 +46,7 @@ end
 %>
 
 <%= include_gon %>
+<%= javascript_include_tag "autocompleter.js" %>
```
Comment (myabc) [inline app/views/work_packages/index.html.erb:49]: what's this for?

**Comment by myabc** [comment]
FYI, getting an error when removing columns:

```
RuntimeError (API Error: No column names):
  app/controllers/api/v3/work_packages_controller.rb:72:in `column_data'
```

File: app/assets/javascripts/angular/helpers/queries-helper.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+angular.module('openproject.workPackages.helpers')
+
+.factory('QueriesHelper', function() {
+  var QueriesHelper = {
+    getAvailableColumns: function (allColumns, columns) {
```
Comment (tilt) [inline app/assets/javascripts/angular/helpers/queries-helper.js:33]: The QueryService already provides these so we should DRY this up

**Comment by tilt** [comment]
would like to integrate this PR with #1302

File: app/assets/javascripts/angular/controllers/dialogs/columns.js [hunk 4]
```diff
@@ -36,7 +36,53 @@ angular.module('openproject.workPackages.controllers')
...
   this.name    = 'Columns';
   this.closeMe = columnsModal.deactivate;
+
+  $scope.getObjectsData = function(term, result) {
+    // Note: This relies on the columns having been cached in the service so they can be instantly available.
+    result($scope.availableColumnsData);
+  };
+
+  // Selected Columns
+  $scope.selectedColumns = QueryService.getSelectedColumns();
```
Comment (tilt) [inline app/assets/javascripts/angular/controllers/dialogs/columns.js:57]: we shouldn't put stuff into the scope unless it's needed

File: app/assets/javascripts/angular/controllers/dialogs/columns.js [hunk 5]
```diff
@@ -36,7 +36,53 @@ angular.module('openproject.workPackages.controllers')
...
+    .map(function(column){ return { id: column.name, label: column.title }; });
+  $scope.previouslySelectedColumnNames = $scope.selectedColumns
+    .map(function(column){ return column.name; });
+
+  // Available Columns
+  QueryService.getAvailableColumns()
+    .then(function(available_columns){
+      $scope.availableColumns = available_columns
+      $scope.availableColumnsData = available_columns.map(function(column){
+        return { id: column.name, label: column.title, other: column.title };
```
Comment (tilt) [inline app/assets/javascripts/angular/controllers/dialogs/columns.js:68]: this duplicates line 59

File: app/assets/javascripts/angular/controllers/dialogs/columns.js [hunk 6]
```diff
@@ -36,7 +36,53 @@ angular.module('openproject.workPackages.controllers')
...
+
+    // Augment work packages with new columns data
+    var addedColumns = $scope.selectedColumns.select(function(column){
+      return $scope.previouslySelectedColumnNames.indexOf(column.name) < 0;
+    });
+    var args = [WorkPackagesTableService.getRowsData(), addedColumns];
+    if (WorkPackagesTableService.getGroupBy().length){
+      args.push(WorkPackagesTableService.getGroupBy());
+    }
+    WorkPackageService.augmentWorkPackagesWithColumnsData.apply(this, args);
```
Comment (tilt) [inline app/assets/javascripts/angular/controllers/dialogs/columns.js:84]: this looks way too clever, I don't believe that the WorkPackageService needs a different `this` context

Comment (tilt) [inline app/assets/javascripts/angular/controllers/dialogs/columns.js:84]: ok sorry, you did this for the optional argument. But we could leave that blank. I've integrated this PR into mine and done a bit of refactoring

**Comment by tilt** [comment]
integrating into #1302

**Pull request closed without merging**